Archives/VolFile.cpp

```diff
@@ -284,42 +279,36 @@ namespace Archives
 		// Write the index table
 		WriteTag(volWriter, volInfo.indexTableLength, "voli");
 
-		volWriter.Write(volInfo.indexEntry, volInfo.indexTableLength);
+		volWriter.Write(volInfo.indexEntries.data(), volInfo.indexTableLength);
 
 		volWriter.Write(&padding, volInfo.paddedIndexTableLength - volInfo.indexTableLength);
 	}
 
 	bool VolFile::OpenAllInputFiles(CreateVolumeInfo &volInfo)
 	{
-		// Allocate space to keep track of all input files
-		volInfo.indexEntry = new IndexEntry[volInfo.fileCount()];
-		volInfo.fileHandle = new HANDLE[volInfo.fileCount()];
-		volInfo.fileNameLength = new int[volInfo.fileCount()];
 		volInfo.stringTableLength = 0;
 
 		// Open all the input files and store indexing info
 		for (std::size_t i = 0; i < volInfo.fileCount(); i++)
 		{
-			volInfo.fileHandle[i] = CreateFileA(
+			HANDLE fileHandle = CreateFileA(
 				volInfo.filesToPack[i].c_str(),	// fileName
 				GENERIC_READ,			// access mode
 				0,						// share mode
 				nullptr,				// security attributes
 				OPEN_EXISTING,			// creation disposition
 				FILE_ATTRIBUTE_NORMAL,	// file attributes
 				nullptr);				// template
+			
+			volInfo.fileHandles.push_back(fileHandle);
 
-			if (volInfo.fileHandle[i] == INVALID_HANDLE_VALUE)// Check for errors
+			if (fileHandle == INVALID_HANDLE_VALUE)// Check for errors
 			{
 				// Error opening file. Close already opened files and return error
 				for (i--; i >= 0; i--) {
```

Comment:
Unrelated to your changes here, but this is an infinite loop for unsigned types. Perhaps the type of `i` was changed to unsigned, or perhaps the bug always existed. We should restructure this code if continued changes don't manage to eliminate it.